proxy/proxy/client.go

```diff
@@ -111,18 +111,15 @@ func (c *Client) Run(connSrc <-chan Conn) {
 }
 
 func (c *Client) handleConn(conn Conn) {
-	// Track connections count only if a maximum connections limit is set to avoid useless overhead
-	if c.MaxConnections > 0 {
-		active := atomic.AddUint64(&c.ConnectionsCounter, 1)
+	active := atomic.AddUint64(&c.ConnectionsCounter, 1)
 
-		// Deferred decrement of ConnectionsCounter upon connection closing
-		defer atomic.AddUint64(&c.ConnectionsCounter, ^uint64(0))
+	// Deferred decrement of ConnectionsCounter upon connection closing
+	defer atomic.AddUint64(&c.ConnectionsCounter, ^uint64(0))
 
-		if active > c.MaxConnections {
-			logging.Errorf("too many open connections (max %d)", c.MaxConnections)
-			conn.Conn.Close()
-			return
-		}
+	if active > c.MaxConnections && c.MaxConnections > 0 {
+		logging.Errorf("too many open connections (max %d)", c.MaxConnections)
```

Comment:
Can we have a different message in the case that we blocked the connection because we are shutting down? Otherwise these logs during shutdown might be a little confusing.

Reply:
Resolved because the "block connections" feature was dropped.